packit/upstream.py

```diff
@@ -627,6 +633,70 @@ def create_rpms(self, rpm_dir: Union[str, Path] = None) -> List[Path]:
         rpms = Upstream._get_rpms_from_rpmbuild_output(out)
         return [Path(rpm) for rpm in rpms]
 
+    def create_rpms_from_srpm(
+        self, srpm: Union[str, Path], rpm_dir: Union[str, Path, None] = None
```

Comment:
Why can't we just update the existing [create_rpms()](https://github.com/packit/packit/blob/525e54049f1219378a8017d803dc7609b77deb4a/packit/upstream.py#L574) to optionally accept a srpm path?

Reply:
there are different switches and directories, but I factored it out, please have a look